### PR TITLE
3.5 usage page

### PR DESCRIPTION
**Pull request by svlandeg**
WIP

## Description
Adding page to the `usage` section describing 3.5 features and (in)compatibilities

### Types of change
docs update

## Checklist
- [x] I confirm that I have the right to submit this contribution under the project's MIT license.
- [x] I ran the tests, and all new and existing tests passed.
- [x] My changes don't require a change to the documentation, or if they do, I've added all required information.

### CODE REVIEW

**Comment by shadeMe** [comment]
Let's not forget the [fuzzy matcher PR](https://github.com/explosion/spaCy/pull/11359).

File: website/docs/usage/v3-5.mdx [hunk 1]
```diff
@@ -0,0 +1,217 @@
...
+```
+
+#### benchmark CLI
+
+The [`benchmark` CLI](/api/cli#benchmark) has been added to extend the existing
+`evaluate` functionality with a wider range of profiling subcommands.
+
+The `benchmark accuracy` CLI is introduced as an alias for `evaluate`.
+
+The new `benchmark speed` CLI performs warmup rounds before measuing the speed
```
Comment (svlandeg) [inline website/docs/usage/v3-5.mdx:34]:
```suggestion
The new `benchmark speed` CLI performs warmup rounds before measuring the speed
```

File: website/docs/usage/v3-5.mdx [hunk 2]
```diff
@@ -0,0 +1,217 @@
...
+```bash
+$ spacy apply en_core_web_sm my_texts/ output.spacy
+```
+
+#### benchmark CLI
+
+The [`benchmark` CLI](/api/cli#benchmark) has been added to extend the existing
+`evaluate` functionality with a wider range of profiling subcommands.
+
+The `benchmark accuracy` CLI is introduced as an alias for `evaluate`.
```
Comment (svlandeg) [inline website/docs/usage/v3-5.mdx:32]: I'd maybe remove a few line breaks, as there's a lot of whitespace in the preview

**Comment by adrianeboyd** [comment]
I'll go ahead and merge this after the tests pass so that it's included in the v3.5.0 tag, but there's no problem to continue making edits before it's published on the website.

**Pull request merged**